### ISSUE TITLE
Launching a game with UMU via Lutris flatpak doesn't work if launched from Steam Deck gaming mode

### ISSUE THREAD

**Issue by pizzadude** [lutris/lutris#5436]
Not sure where to report this, so I'll report it on both umu-launcher and lutris github.

If I try to launch a game that is installed via Lutris, (which I have switched the prefix to use GE-Proton (which uses umu) from previously Wine-GE)) in the Lutris flatpak, it works fine.

It even launches fine if ran as a non-steam game through Steam. (lutris flatpak with the game's id is added as non steam game, I used Boilr for this).

However, if I launch the game from gaming mode steam instead of desktop mode (on a Steam Deck), the game doesn't launch at all.

Switching the prefix back to Wine-GE, the problem goes away.

Any ideas?

**Comment by bitstrom** [comment]
I have the same problem on my Steam Deck with umu. Games installed through Lutris start fine from Lutris or Steam in desktop mode, but crash in gaming mode on start up.

Desktop mode Proton log: 
```
======================
Proton: 1711108288 proton-9.0-beta16
SteamGameId: default
Command: ['start', '/unix', '/run/media/deck/11953d42-3564-4292-a560-40084f4678a2/Games/slave-zero/drive_c/users/Public/Desktop/Slave Zero.lnk']
Options: {'forcelgadd'}
depot: 0.20240125.75305
pressure-vessel: 0.20240125.0 scout
scripts: 0.20240125.0
sniper: 0.20240125.75305 sniper 0.20240125.75305
Kernel: Linux 6.1.52-valve16-1-neptune-61 #1 SMP PREEMPT_DYNAMIC Tue, 06 Feb 2024 00:51:49 +0000 x86_64
Language: LC_ALL None, LC_MESSAGES None, LC_CTYPE None
System WINEDLLOVERRIDES: winemenubuilder=
Effective WINEDLLOVERRIDES: winemenubuilder=
System WINEDEBUG: 
Effective WINEDEBUG: 
======================
wine: using kernel write watches (experimental).
fsync: up and running.
wine: using kernel write watches (experimental).
wine: using kernel write watches (experimental).
wine: using kernel write watches (experimental).
wine: using kernel write watches (experimental).
0070:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0070:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0070:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0070:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0088:err:hid:udev_bus_init UDEV monitor creation failed
007c:fixme:wineusb:add_usb_device Interface 1 has 6 alternate settings; using the first one.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
0090:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0090:fixme:xinput:pdo_pnp IRP_MN_QUERY_ID type 5, not implemented!
0090:fixme:xinput:pdo_pnp IRP_MN_QUERY_ID type 5, not implemented!
0090:fixme:xinput:pdo_pnp IRP_MN_QUERY_ID type 5, not implemented!
wine: using kernel write watches (experimental).
0090:fixme:xinput:pdo_pnp IRP_MN_QUERY_ID type 5, not implemented!
wine: using kernel write watches (experimental).
wine: using kernel write watches (experimental).
wine: using kernel write watches (experimental).
wine: using kernel write watches (experimental).
0114:fixme:oleacc:find_class_data unhandled window class: L"#32769"
0114:fixme:uiautomation:uia_get_providers_for_hwnd Override provider callback currently unimplemented.
0114:fixme:uiautomation:default_uia_provider_callback Default ProviderType_NonClientArea provider unimplemented.
[S_API] SteamAPI_Init(): SteamAPI_IsSteamRunning() did not locate a running instance of Steam.
[S_API FAIL] SteamAPI_Init() failed
0024:err:steam:main SteamAPI_Init failed
0024:err:steam:setup_steam_files Could not convert "" to win path.
0114:fixme:oleacc:find_class_data unhandled window class: L"#32769"
0114:fixme:uiautomation:base_hwnd_fragment_get_FragmentRoot 0000000001349BE8, 000000000021F9D0: stub!
0114:fixme:uiautomation:base_hwnd_fragment_GetEmbeddedFragmentRoots 0000000001349BE8, 000000000021F9E0: stub!
wine: using kernel write watches (experimental).
wine: using kernel write watches (experimental).
0130:fixme:exec:SHELL_execute flags ignored: 0x00000100
0130:fixme:exec:SHELL_execute flags ignored: 0x00004100
wine: using kernel write watches (experimental).
0144:fixme:ddraw:DirectDrawEnumerateExA flags 0x4 not handled
/usr/lib/i386-linux-gnu/GL/default/share/libdrm/amdgpu.ids: No such file or directory
/usr/lib/i386-linux-gnu/GL/default/share/libdrm/amdgpu.ids: No such file or directory
0144:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
0144:fixme:x11drv:window_update_fshack Enabling fshack for hwnd 0x3007e, mapping virt window (0,0)-(1024,768), client (0,0)-(1024,768) to host window (0,0)-(1280,800), client (0,0)-(1280,800).
0144:err:ole:com_get_class_object class {5959df60-2911-11d1-b049-0020af30269a} not registered
0144:err:ole:com_get_class_object no class object {5959df60-2911-11d1-b049-0020af30269a} could be created for context 0x1
0144:fixme:pulse:pulse_channel_map_to_channel_mask Unhandled channel aux0
0144:fixme:pulse:pulse_channel_map_to_channel_mask Unhandled channel aux1
0144:err:ole:apartment_add_dll couldn't load in-process dll L"a3d.dll"
0144:err:ole:com_get_class_object no class object {442d12a1-2641-11d2-90fb-006008a1f441} could be created for context 0x1
0124:fixme:uiautomation:msaa_provider_GetPatternProvider Unimplemented patternId 10002
0124:fixme:uiautomation:base_hwnd_provider_GetPatternProvider 0000000001389C20, 10002, 0000000001A2F8A0: stub
0144:fixme:ddraw:ddraw_surface1_Flip Ignoring flags 0x1.
0154:fixme:d3d:state_linepattern_w Setting line patterns is not supported in OpenGL core contexts.
msg:gl_debug_message_callback called from native thread, severity 0x826b, message "Shader Stats: SGPRS: 24 VGPRS: 16 Code Size: 292 LDS: 0 Scratch: 0 Max Waves: 16 Spilled SGPRs: 0 Spilled VGPRs: 0 PrivMem VGPRs: 0 Outputs: 1 PatchOutputs: 0 DivergentLoop: 0 InlineUniforms: 0 (PS, W64)".
0144:fixme:x11drv:window_update_fshack Disabling fshack for hwnd 0x3007e, mapping virt window (0,0)-(1280,800), client (4,23)-(1276,796) to host window (0,0)-(1280,800), client (0,0)-(1016,741).
```

Gaming mode Proton log: 
```
======================
Proton: 1711108288 proton-9.0-beta16
SteamGameId: default
Command: ['start', '/unix', '/run/media/deck/11953d42-3564-4292-a560-40084f4678a2/Games/slave-zero/drive_c/users/Public/Desktop/Slave Zero.lnk']
Options: {'forcelgadd'}
depot: 0.20240125.75305
pressure-vessel: 0.20240125.0 scout
scripts: 0.20240125.0
sniper: 0.20240125.75305 sniper 0.20240125.75305
Kernel: Linux 6.1.52-valve16-1-neptune-61 #1 SMP PREEMPT_DYNAMIC Tue, 06 Feb 2024 00:51:49 +0000 x86_64
Language: LC_ALL None, LC_MESSAGES None, LC_CTYPE None
System WINEDLLOVERRIDES: winemenubuilder=
Effective WINEDLLOVERRIDES: winemenubuilder=
System WINEDEBUG: 
Effective WINEDEBUG: 
======================
wine: using kernel write watches (experimental).
fsync: up and running.
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
0070:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0070:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0070:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0070:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0088:err:hid:udev_bus_init UDEV monitor creation failed
007c:fixme:wineusb:add_usb_device Interface 1 has 6 alternate settings; using the first one.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
00d0:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
00d0:err:winediag:nodrv_CreateWindow L"The explorer process failed to start."
00d0:err:systray:initialize_systray Could not create tray window
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
[S_API] SteamAPI_Init(): SteamAPI_IsSteamRunning() did not locate a running instance of Steam.
[S_API FAIL] SteamAPI_Init() failed
0024:err:steam:main SteamAPI_Init failed
0024:err:steam:setup_steam_files Could not convert "" to win path.
0110:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
0110:err:winediag:nodrv_CreateWindow L"Make sure that your X server is running and that $DISPLAY is set correctly."
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
0118:fixme:exec:SHELL_execute flags ignored: 0x00000100
0118:fixme:exec:SHELL_execute flags ignored: 0x00004100
0124:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
0124:err:winediag:nodrv_CreateWindow L"Make sure that your X server is running and that $DISPLAY is set correctly."
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
012c:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
012c:err:winediag:nodrv_CreateWindow L"Make sure that your X server is running and that $DISPLAY is set correctly."
012c:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
wine: Unhandled page fault on read access to 00000000 at address 00000000 (thread 012c), starting debugger...
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
0134:fixme:dbghelp:elf_search_auxv can't find symbol in module
Unhandled exception: page fault on read access to 0x00000000 in wow64 32-bit code (0x00000000).
0134:fixme:dbghelp:elf_search_auxv can't find symbol in module
Register dump:
 CS:0023 SS:002b DS:002b ES:002b FS:0063 GS:006b
 EIP:00000000 ESP:0082f76c EBP:0082feb4 EFLAGS:00010207(  R- --  I   - -P-C)
 EAX:004e3268 EBX:004e3268 ECX:0082f7b4 EDX:00000000
 ESI:0082f7b4 EDI:0041d010
Stack dump:
0x0082f76c:  0041d02c 004e3268 0082f7b4 0043555b
0x0082f77c:  004e3268 0082f7b4 00000000 00000000
0x0082f78c:  003f1000 0041d0a6 0041d010 004e3268
0x0082f79c:  0082f7b4 00405f54 00000438 0082f7b4
0x0082f7ac:  00000000 00000000 00000438 00000000
0x0082f7bc:  0040a335 0082f7cc 0082f7e2 0015b6f0
Backtrace:
=>0 0x00000000 (0x0082feb4)
  1 0x004440c6 in slavezero (+0x440c6) (0x0082ff50)
  2 0x7bece97c in kernel32 (+0xe97c) (0x0082ff68)
  3 0x7bf4cb13 in ntdll (+0xcb13) (0x0082ff80)
  4 0x7bf7ebc2 in ntdll (+0x3ebc2) (0x0082ffec)
0x00000000: -- no code accessible --
Modules:
Module  Address                 Debug info      Name (54 modules)
PE        400000-  52d000       Export          slavezero
PE      10000000-1000a000       Deferred        mrgplay
PE-Wine 7b1b0000-7b1e6000       Deferred        uxtheme
PE-Wine 7b200000-7b21c000       Deferred        imm32
PE-Wine 7b230000-7b25b000       Deferred        msvcirt
PE-Wine 7b270000-7b298000       Deferred        ws2_32
PE-Wine 7b2b0000-7b2bb000       Deferred        nsi
PE-Wine 7b2d0000-7b2e4000       Deferred        dnsapi
PE-Wine 7b300000-7b324000       Deferred        iphlpapi
PE-Wine 7b340000-7b34f000       Deferred        wsock32
PE-Wine 7b360000-7b381000       Deferred        msacm32
PE-Wine 7b3a0000-7b464000       Deferred        winmm
PE-Wine 7b480000-7b493000       Deferred        coml2
PE-Wine 7b4b0000-7b535000       Deferred        rpcrt4
PE-Wine 7b550000-7b59d000       Deferred        combase
PE-Wine 7b5b0000-7b6c4000       Deferred        ole32
PE-Wine 7b6e0000-7b710000       Deferred        win32u
PE-Wine 7b720000-7b7a7000       Deferred        gdi32
PE-Wine 7b7c0000-7b8a2000       Deferred        ucrtbase
PE-Wine 7b8c0000-7b8e8000       Deferred        sechost
PE-Wine 7b900000-7b9b2000       Deferred        msvcrt
PE-Wine 7b9d0000-7ba12000       Deferred        advapi32
PE-Wine 7ba30000-7bbe2000       Deferred        user32
PE-Wine 7bc00000-7bea3000       Deferred        kernelbase
PE-Wine 7bec0000-7bf25000       COFF            kernel32
PE-Wine 7bf40000-7bfed000       COFF            ntdll
ELF     f5f5b000-f5f84000       Deferred        libgcc_s.so.1
ELF     f5f84000-f5f8e000       Deferred        libuuid.so.1
ELF     f5f8e000-f5fba000       Deferred        libexpat.so.1
ELF     f5fba000-f6009000       Deferred        libfontconfig.so.1
ELF     f6009000-f602c000       Deferred        libbrotlicommon.so.1
ELF     f602c000-f603a000       Deferred        libbrotlidec.so.1
ELF     f603a000-f6054000       Deferred        libz.so.1
ELF     f6054000-f6094000       Deferred        libpng16.so.16
ELF     f6094000-f615b000       Deferred        libfreetype.so.6
ELF     f6166000-f6178000       Deferred        libresolv.so.2
ELF     f617d000-f6183000       Deferred        ws2_32.so
ELF     f6183000-f6188000       Deferred        dnsapi.so
ELF     f6188000-f62e5000       Deferred        win32u.so
ELF     f62e9000-f63cb000       Export          ntdll.so
ELF     f78a5000-f79ff000       Deferred        libx11.so.6
ELF     f79ff000-f7a3a000       Deferred        libglx.so.0
ELF     f7a3a000-f7ab6000       Deferred        libgldispatch.so.0
ELF     f7ab6000-f7b8f000       Deferred        libm.so.6
ELF     f7b8f000-f7c00000       Deferred        libgl.so.1
ELF     f7c00000-f7e32000       Deferred        libc.so.6
ELF     f7e36000-f7e3c000       Deferred        libxau.so.6
ELF     f7e3c000-f7e6c000       Deferred        libxcb.so.1
ELF     f7e6e000-f7e73000       Deferred        librt.so.1
ELF     f7e73000-f7e78000       Deferred        libdl.so.2
ELF     f7e78000-f7e7d000       Deferred        libpthread.so.0
ELF     f7e88000-f7ee0000       Deferred        gameoverlayrenderer.so
ELF     f7ee2000-f7f17000       Deferred        ld-linux.so.2
ELF     f7f17000-f7f1c000       Deferred        <wine-loader>
Threads:
process  tid      prio    name (all IDs are in hex)
00000020 steam.exe
	00000024    0     
	00000110    0     
	0000011c    0     
00000030 services.exe
	00000034    0     
	00000038    0     wine_rpcrt4_server
	00000044    0     wine_rpcrt4_io
	00000048    0     wine_threadpool_worker
	00000054    0     wine_rpcrt4_io
	0000006c    0     wine_rpcrt4_io
	00000078    0     wine_rpcrt4_io
	0000009c    0     wine_rpcrt4_io
	000000b4    0     wine_rpcrt4_io
	000000c8    0     
	000000e8    0     wine_rpcrt4_io
	000000ec    0     wine_threadpool_worker
0000003c winedevice.exe
	00000040    0     
	0000004c    0     
	00000050    0     wine_sechost_service
	00000058    0     
	0000005c    0     
	00000060    0     
	000000c0    0     
	000000c4    0     
00000064 winedevice.exe
	00000068    0     
	00000070    0     
	00000074    0     wine_sechost_service
	0000007c    0     
	00000080    0     
	00000084    0     
	00000090    0     
00000094 plugplay.exe
	00000098    0     
	000000a0    0     
	000000a4    0     wine_sechost_service
	000000a8    0     wine_rpcrt4_server
000000ac svchost.exe
	000000b0    0     
	000000b8    0     
	000000bc    0     wine_sechost_service
000000e0 rpcss.exe
	000000e4    0     
	000000f0    0     
	000000f4    0     wine_sechost_service
	000000f8    0     wine_rpcrt4_server
	000000fc    0     wine_rpcrt4_server
	00000104    0     wine_threadpool_worker
00000128 (D) C:\MagiPacks\Slave Zero\SlaveZero.exe
	0000012c    0 <== 
	00000140    0     
00000138 conhost.exe
	0000013c    0     
System information:
    Wine build: wine-9.0
    Platform: x86_64 (guest: i386)
    Version: Windows 10
    Host system: Linux
    Host version: 6.1.52-valve16-1-neptune-61
pid 158 != 157, skipping destruction (fork without exec?)
```

**Comment by ace3** [comment]
@pizzadude , hi , ho do you switch back to Wine-GE ?

i'm having the same issue.

**Comment by R1kaB3rN** [comment]
@strycore feel free to close this issue. 

This issue is a duplicate of https://github.com/Open-Wine-Components/umu-launcher/issues/92 and the problem has already been fixed upstream at https://github.com/Open-Wine-Components/umu-launcher/commit/6ff8c44b52f33d89b2e3b06cd33e3f6ec70c3318.

**Comment by AlmeidaRW** [comment]
Issue still persists even after Lutris 0.5.18 Flatpak has been released. Games running under ge-proton runner do not work in Gaming mode, only Desktop. Games running under wine-ge work correctly in both modes.

**Comment by R1kaB3rN** [comment]
To be clear, the reason this issue exists for users in Lutris Flatpak 0.5.18 within Steam Deck's gaming mode is due to https://github.com/lutris/lutris/commit/6b745405f028f4f7e4483f94fbde4432c7100774. Since the `run` verb is used, the [code path related to bringing the game's window to the foreground will not be reached](https://github.com/Open-Wine-Components/umu-launcher/blob/6189d0d9fd062e89a375db20aeae1d1c009e9833/umu/umu_run.py#L658). The current workaround for now is to set PROTON_VERB=waitforexitandrun until the Lutris Flatpak has been updated to at least https://github.com/lutris/lutris/commit/d71a2d41525d4f3accf745fea5d303882890123c.

**Comment by pizzadude** [comment]
> To be clear, the reason this issue exists for users in Lutris Flatpak 0.5.18 within Steam Deck's gaming mode is due to [6b74540](https://github.com/lutris/lutris/commit/6b745405f028f4f7e4483f94fbde4432c7100774). Since the `run` verb is used, the [code path related to bringing the game's window to the foreground will not be reached](https://github.com/Open-Wine-Components/umu-launcher/blob/6189d0d9fd062e89a375db20aeae1d1c009e9833/umu/umu_run.py#L658). The current workaround for now is to set `PROTON_VERB=waitforexitandrun` until the Lutris Flatpak has been updated to at least [d71a2d4](https://github.com/lutris/lutris/commit/d71a2d41525d4f3accf745fea5d303882890123c).

I tried setting that environment variable with both flatseal and within the individual game's lutris settings but it did not fix the issue.

**Comment by AlmeidaRW** [comment]
> > To be clear, the reason this issue exists for users in Lutris Flatpak 0.5.18 within Steam Deck's gaming mode is due to [6b74540](https://github.com/lutris/lutris/commit/6b745405f028f4f7e4483f94fbde4432c7100774). Since the `run` verb is used, the [code path related to bringing the game's window to the foreground will not be reached](https://github.com/Open-Wine-Components/umu-launcher/blob/6189d0d9fd062e89a375db20aeae1d1c009e9833/umu/umu_run.py#L658). The current workaround for now is to set `PROTON_VERB=waitforexitandrun` until the Lutris Flatpak has been updated to at least [d71a2d4](https://github.com/lutris/lutris/commit/d71a2d41525d4f3accf745fea5d303882890123c).
> 
> I tried setting that environment variable with both flatseal and within the individual game's lutris settings but it did not fix the issue.

I set the environment variable in the System options tab of the Game Configuration and it did work for me.

**Comment by rollercoaster** [comment]
> The current workaround for now is to set `PROTON_VERB=waitforexitandrun` until the Lutris Flatpak has been updated to at least [d71a2d4](https://github.com/lutris/lutris/commit/d71a2d41525d4f3accf745fea5d303882890123c).

Hi, do you know if that update has been pushed to release and distribution ?

**Comment by danieljohnson2** [comment]
No, that commit [d71a2d4](https://github.com/lutris/lutris/commit/d71a2d41525d4f3accf745fea5d303882890123c) went in just after 0.5.18 was released.

**Comment by strycore** [comment]
We need to release 0.5.19 asap.

**Comment by danieljohnson2** [comment]
I see no reason to wait on 0.5.19 at this point. I was looking into #5899, but I think that is a server-side data problem, and so not an issue for a release.

If you want me to write the changelog, let me know. Looks like it is mainly bug-fixes, including several for Proton support. 

The refactored widget generator should be in decent shape now, but its bugs have been mostly cosmetic, so far.

**Comment by strycore** [comment]
Yes, we can start writing the changelog, start a round of testing and maybe get a few more bugfixes shipped (the new BIOS autodetection may still have bugs for example)

#5899 can likely be worked around by restarting lutris, I think this happens because the dropdowns get populated while the client is still downloading updates.

**Comment by danieljohnson2** [comment]
#5899 isn't fixed by restarting. I think ‘’versions.json’’ and ’’dxvk_versions.json’’ disagree, actually.

**Comment by strycore** [comment]
`versions.json` is right and `dxvk_versions.json` is wrong, we hate and we want to delete.

**Comment by danieljohnson2** [comment]
``versions.json`` does not have a list of versions for the drop down. Perhaps it should, but we have what we have.

**Comment by strycore** [comment]
It's not meant to have a list of versions. In the Lutris API, we have the latest version available. Github releases has the list of all versions created. 

Downloading and using the latest version of DXVK is a top priority Lutris feature whereas having the opportunity to switch between DXVK version is something we offer because we're nice and don't want to prevent users from tinkering and breaking their computers.

**Comment by danieljohnson2** [comment]
Oh. That's a little awkward- Lutris will default to the latest 1.x version of DXVK if you don't have Vulkan 1.3 or later, and without that list, that trick won't work.

Maybe that's too marginal to be a worry. Should we remove the not-installed versions of runtimes from that drop down, except the default from ``versions.json``?

**Comment by strycore** [comment]
> Oh. That's a little awkward- Lutris will default to the latest 1.x version of DXVK if you don't have Vulkan 1.3 or later, and without that list, that trick won't work.

No it should work, or at least partially. That's why we send the PCI IDs to the server. If the PCI ID match a GPU that doesn't support Vulkan 1.3, we replace DXVK 2 by DXVK 1 in versions.json

**Comment by danieljohnson2** [comment]
That makes the client side check even more marginal, indeed.

**Comment by rollercoaster** [comment]
> We need to release 0.5.19 asap.
The version is up on gitbut but not on discover on steamdeck.
Any way to manually install the update?
